sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -956,11 +957,13 @@ public boolean advance() throws IOException {
               rawRecord.topic(),
               rawRecord.partition(),
               rawRecord.offset(),
+              consumerSpEL.getEventTimestamp(rawRecord),
               decode(rawRecord.key(), source.spec.getKeyCoder()),
               decode(rawRecord.value(), source.spec.getValueCoder()));
 
           curTimestamp = (source.spec.getTimestampFn() == null)
```

Comment:
combined with previous comment, after this change, no matter it's Kafka0.9 or 0.10, a `timestamp` field is there.
as Kafka0.9 does't have a method `KafkaRecord.getTimestamp()`, I move the logic to `consumerSpEL.getEventTimestamp(rawRecord)`, then
* For kafka0.9, `hasEventTimestamp==false`, so `return System.currentTimeMillis();` as before;
* for Kakfa0.10,  `hasEventTimestamp==true`, if `NO_TIMESTAMP` returns `System.currentTimeMillis();`, otherwise return `KafkaRecord.getTimestamp()`. IMO -1 is not good as EventTimestamp.